### ISSUE TITLE
Different preview links in multi domain installations

### ISSUE THREAD

**Issue by OMOSde** [contao/contao#5972]
### Affected version(s)

4.13.20

### Description

Creating the preview links does not work correctly in multi-domain installations. Apparently the generation of the link is based on the domain you are currently logged in with in the backend.

Only one of the links worked in the frontend then.

Reproduce:

- Create 2 starting points with registered domains
- create one preview link per domain
- login with each domain once in the backend and copy preview links (the links are different)

**Comment by aschempp** [comment]
can you please explain what "different" means?

**Comment by OMOSde** [comment]
If I remember correctly, the generated link was prefixed with the domain you are currently logged into the system with.

**Comment by fritzmg** [comment]
That should only be the case for links to pages that do not have a domain set in their website root's settings.

**Comment by OMOSde** [comment]
I have now tested this in a live environment. (Contao 4.13.28 - PHP 8.1.21)

I have 2 starting points with different domains in their website root settings.

I logged in with the 1st domain in the backend and created a preview link via preview.php.

The function `copy to clipboard` results in a link: https://www.domain-a.de/_contao/preview/5?_hash=xxxTokenxxx

After that I logged out and logged into the same backend using the 2nd domain.

I use for the same preview link again the function `copy to clipboard` and get as link: https://www.domain-b.de/_contao/preview/5?_hash=yyyTokenyyy

The two tokens are not identical.
